convert2rhel/actions/__init__.py

```diff
@@ -0,0 +1,784 @@
+import os.path
+import pkgutil
+import re
+
+from functools import cmp_to_key, wraps
+
+import six
+
+from convert2rhel import grub, pkgmanager, utils
+from convert2rhel.pkghandler import (
+    call_yum_cmd,
+    compare_package_versions,
+    get_installed_pkg_objects,
+    get_pkg_fingerprint,
+    get_total_packages_to_update,
+)
+from convert2rhel.repo import get_hardcoded_repofiles_dir
+from convert2rhel.systeminfo import system_info
+from convert2rhel.toolopts import tool_opts
+from convert2rhel.utils import ask_to_continue, get_file_content, run_subprocess
+
+
+logger = logging.getLogger(__name__)
+
+KERNEL_REPO_RE = re.compile(r"^.+:(?P<version>.+).el.+$")
+KERNEL_REPO_VER_SPLIT_RE = re.compile(r"\W+")
+BAD_KERNEL_RELEASE_SUBSTRINGS = ("uek", "rt", "linode")
+
+RPM_GPG_KEY_PATH = os.path.join(utils.DATA_DIR, "gpg-keys", "RPM-GPG-KEY-redhat-release")
+# The SSL certificate of the https://cdn.redhat.com/ server
+SSL_CERT_PATH = os.path.join(utils.DATA_DIR, "redhat-uep.pem")
+CDN_URL = "https://cdn.redhat.com/content/public/convert2rhel/$releasever/$basearch/os/"
+CONVERT2RHEL_REPO_CONTENT = """\
+[convert2rhel]
+name=Convert2RHEL Repository
+baseurl=%s
+gpgcheck=1
+enabled=1
+sslcacert=%s
+gpgkey=file://%s""" % (
+    CDN_URL,
+    SSL_CERT_PATH,
+    RPM_GPG_KEY_PATH,
+)
+
+PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
+
+LINK_KMODS_RH_POLICY = "https://access.redhat.com/third-party-software-support"
+LINK_PREVENT_KMODS_FROM_LOADING = "https://access.redhat.com/solutions/41278"
+# The kernel version stays the same throughout a RHEL major version
+COMPATIBLE_KERNELS_VERS = {
+    7: "3.10.0",
+    8: "4.18.0",
+}
+
+#: Status code of an Action.
+#:
+#: When an action completes, it may have succeeded or failed.  We set the
+#: `Action.status` attribute to one of the following values so that we know
+#: what happened.  This mapping lets us use a symbolic name for the status
+#: for readability but that is mapped to a specific integer for consumption
+#: by other tools.
+#:
+#: .. note:: At the moment, we only make use of SUCCESS and ERROR.  Other
+#:      statuses may be used in future releases as we refine this system
+#:      and start to use it with console.redhat.com
+#:
+#: :SUCCESS: no problem.
+#: :WARNING: the problem is just a warning displayed to the user. (unused,
+#:      warnings are currently emitted directly from the Action)
+#: :OVERRIDABLE: the error caused convert2rhel to fail but the user has
+#:      the option to ignore the check in a future run.
+#: :ERROR: the error caused convert2rhel to fail the conversion, but further
+#:      tests can be run.
+#: :FATAL: the error caused convert2rhel to stop immediately.
+#:
+#: .. warning:: Do not change the numeric value of these statuses once they
+#:      have been in a public release as external tools may be depending on
+#:      the value.
+STATUS_CODE = {
+    "SUCCESS": 0,
+    "WARNING": 300,
+    "OVERRIDABLE": 600,
+    "ERROR": 900,
+    "FATAL": 1200,
+}
```

Comment:
Rescale this to range from 0-255 since we may use this with the exit code of /usr/bin/convert2rhel as well

Reply:
Done